### PR TITLE
proxy

### PR DESCRIPTION
**Pull request by akamarouski**

### CODE REVIEW

File: roles/devices/templates/zebrunner-farm [hunk 1]
```diff
@@ -66,6 +66,26 @@ function create_containers () {
     local wireless=`cat ${devices} | grep "$udid" | cut -d '|' -f 9`
     #echo "wireless: $wireless"
 
+    local server_proxy_port=`cat ${devices} | grep "$udid" | cut -d '|' -f 12`
+    #echo "server_proxy_port: $server_proxy_port"
+
+    local certificates_proxy_path={{ CA_CERTS_PROXY_PATH }}
```
Comment (vdelendik) [inline roles/devices/templates/zebrunner-farm:72]: let's try to remove CA_CERTS_PROXY_PATH  usage and switch to the docker volume which is shared across all mitm containers.
item to test: how ca certificates generated/regenerated during 2nd, 3rd etc device start

File: roles/devices/templates/zebrunner-farm [hunk 2]
```diff
@@ -99,6 +119,16 @@ function create_containers () {
     fi
 
   if [ $platform_name == "android" ]; then
+
+      if [ "$server_proxy_port" != "0" ] && [ "$proxy_port" != "0" ]; then
+        docker run -itd --name device-${device_name}-${udid}-mitm -m 1g --log-opt max-size=1024m --log-opt max-file=2 --net=${net} \
+          --restart on-failure \
+          -v ${certificates_proxy_path}:/tmp/zebrunner/mitmproxy \
```
Comment (vdelendik) [inline roles/devices/templates/zebrunner-farm:126]: reuse volume like:
```
-v mcloud-mitm-volume:/tmp/zebrunner/mitmproxy \
```

File: roles/devices/vars/main.yml.original [hunk 3]
```diff
@@ -49,6 +49,9 @@ WDA_FILE:
 # Override to use custom WDA bundle id: https://github.com/zebrunner/mcloud-agent#prepare-webdriveragentipa-file
 WDA_BUNDLEID:
 
+# Path to the folder with CA Authority mitmproxy certificates
+CA_CERTS_PROXY_PATH:
```
Comment (vdelendik) [inline roles/devices/vars/main.yml.original:53]: remove usage

Comment (akamarouski) [reply roles/devices/vars/main.yml.original]: вщту

Comment (akamarouski) [reply roles/devices/vars/main.yml.original]: done

File: roles/devices/templates/zebrunner-farm (hunk 2 above)
Comment (akamarouski) [reply roles/devices/templates/zebrunner-farm]: done

File: roles/devices/templates/zebrunner-farm (hunk 1 above)
Comment (akamarouski) [reply roles/devices/templates/zebrunner-farm]: done

File: roles/devices/templates/zebrunner-farm [hunk 4]
```diff
@@ -99,6 +106,16 @@ function create_containers () {
     fi
 
   if [ $platform_name == "android" ]; then
+
+      if [ "$server_proxy_port" != "0" ] && [ "$proxy_port" != "0" ]; then
+        docker run -itd --name device-${device_name}-${udid}-mitm -m 1g --log-opt max-size=1024m --log-opt max-file=2 --net=${net} \
+          --restart on-failure \
+          -v mcloud-mitm-volume:/opt/mitmproxy \
+          -e PROXY_PERMANENT_ARGS="$proxy_permament_arguments"  \
+          --publish ${server_proxy_port}:8060 -p ${proxy_port}:8080 \
```
Comment (vdelendik) [inline roles/devices/templates/zebrunner-farm:115]: let's reuse widely common `-p` instead of `--publish`

Comment (akamarouski) [reply roles/devices/templates/zebrunner-farm]: done

**Pull request merged**